internal/fs/dir_handle.go

```diff
@@ -26,15 +28,31 @@ import (
 	"golang.org/x/net/context"
 )
 
+// Record stores metadata of dirent entries fetched from GCSfuse
+type Record struct {
+	sync.Mutex
+	length          int
+	err             error
+	entryForSorting fuseutil.Dirent
+	cond            *sync.Cond
+}
+
+func NewRecord() *Record {
+	r := Record{}
+	r.cond = sync.NewCond(&r)
+	return &r
+}
+
 var ContinuationToken string
+var rec = NewRecord()
```

Comment:
why is this a global variable. why can't this be at ReadDir level?

Reply:
Updated in new implementation.